javalib/src/main/scala/java/lang/String.scala

```diff
@@ -560,28 +560,15 @@ final class _String()
     } else if (replacement == null) {
       throw new NullPointerException("replacement should not be null")
     } else {
+      if (target.length == 0) return this
+
       val ts    = target.toString
       var index = indexOf(ts, 0)
 
       if (index == -1) return this
 
       val rs = replacement.toString
 
-      if ("".equals(rs)) {
-        val buffer =
-          new java.lang.StringBuilder(count + (rs.length * (count + 1)))
```

Comment:
This special case is bogus given that `rs.length == 0` so this would be `StringBuilder(count)` and below the `buffer.append(rs)` is a noop. Is this optimization needed?